components/admin_console/plugin_management/plugin_management.tsx

```diff
@@ -700,12 +700,17 @@ export default class PluginManagement extends AdminSettings<Props, State> {
     handleRemove = async () => {
         this.setState({lastMessage: null, serverError: null});
         if (this.state.removing !== null) {
-            const {error} = await this.props.actions.removePlugin(this.state.removing);
-            this.setState({removing: null});
+            const {disableError} = await this.props.actions.disablePlugin(this.state.removing)
```

Comment:
I don't think the bug in https://github.com/mattermost/mattermost-server/issues/19108 is caused by a wrong state on the server side i.e. the plugin is in fact disabled after an uninstall. It's more of an sync issue between the webapp and server. If you take a look at http://localhost:8065/api/v4/plugins/statuses after the plugin has been re-installed, it states that the plugin is in fact disabled i.e. has `state` `0`.

Reply:
@hanzei
Thanks for your input! I wasn't aware that this calls the API. I updated the code with calling the dispatch with `DISABLE*_PLUGIN*` when `removePlugin` is called in the client to update the status.

Reply:
Sounds reasonable to me :+1: